packages/KInput/KInput.vue

```diff
@@ -2,24 +2,35 @@
   <div class="k-input-wrapper">
     <input
       v-if="!label"
-      :value="value"
       v-bind="attrs"
+      :value="currValue ? currValue : value"
+      :class="`k-input-${size}`"
       class="form-control k-input"
-      @input="e => $emit('input', e.target.value)"
+      @input="e => {
+        $emit('input', e.target.value),
+        currValue = e.target.value
+      }"
       v-on="listeners">
 
     <div
       v-else
+      :class="`k-input-label-${size}`"
       class="col-md-4 mt-5">
       <div class="text-on-input">
-        <label :class="{ focused: isFocused }">
+        <label :class="{ focused: isFocused, hovered: isHovered }">
```

Comment:
The label here is missing the `for` attr. I was playing with a library called `vue-unique-id` to handle cases like this. The other option, that I think also helps simplify the hover and focus styling is wrapping the label around the input. In this case, it doesn't require an id/for to associate the label and input.

Reply:
Updated it so if there is a label passed in we are now using `for` attribute.